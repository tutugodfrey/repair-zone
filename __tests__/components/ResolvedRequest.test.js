import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import ConnectResolvedRequest, { ResolvedRequests, mapStateToProps }
  from '../../client/src/components/ResolvedRequests.jsx';

let userData = {
  id: 2,
  fullname: 'fullname',
  isAdmin: true,
}
const initialState = {
  requestReducer: {},
  userReducer: {},
  pageReducer: {},
}
let mockStore = configureStore();
let store = mockStore(initialState);


describe('<ResolvedRequest />', () => {
  describe('unconnected component test', () => {
    let wrapper;
    let requests = [
      {
        user: {
          fullname: 'fullname',
          isAdmin: false,
        },
        request: {
          id: 1,
          status: 'resolved',
        },
      }
    ];
    const dispatch = jest.fn();
    beforeEach(() => {
      wrapper = shallow(
        <ResolvedRequests
          requests={requests}
          userData={userData}
          dispatch={dispatch}
        />
      );
    });
    test('should map the snapshot for the component', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
  describe('connected component test', () => {
    let wrapper;
    let requests = [
      {
        user: {
          fullname: 'fullname',
          isAdmin: false,
        },
        request: {
          id: 1,
          issueData: '2018-06-17T23:00:00.000Z',
          category: 'electrical',
          description: 'my wall socket got burnt and need replacement',
          address: 'provide for user making request',
          urgent: true,
          adminId: 2,
          status: 'Resolved',
          updatedAt: '2018-06-17T23:00:00.000Z'
        },
      }
    ];
    const userData = {
      id: 2,
      fullname: {},
      token: 'tokenvalue',
    }
    store.getState().requestReducer.requests = requests;
    store.getState().userReducer.userData = userData;
    beforeEach(() => {
      wrapper = mount(
        <Provider store={store}>
          <ConnectResolvedRequest />
        </Provider>
      );
    });
    test('should not have a button', () => {
      const btns = wrapper.find('button');
      expect(btns.length).toBe(1)
    })
  });
  describe('mapStateToProps', () => {
    test('should map the state of the component correctly', () => {
      const appState = {
        userReducer: {
          userData: {},
        },
        requestReducer: {
          requests: [],
        },
        pageReducer: {
          errorStatus: true,
          errorMessage: 'ssssss',
        },
      };
      const expectedState = {
        userData: {},
        requests: [], 
      };
      const componentState = mapStateToProps(appState);
      expect(componentState).toEqual(expectedState);
    });
  });
});
