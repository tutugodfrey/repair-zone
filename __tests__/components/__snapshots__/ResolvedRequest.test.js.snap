// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResolvedRequest /> unconnected component test should map the snapshot for the component 1`] = `
<div>
  <Connect(Modal) />
  <div>
    <h1
      className="text-center pt-5"
    >
      Resolved Requests
    </h1>
    <Div
      content={
        Array [
          <div
            className="view-request-div col-10 offset-1 p-0 mb-1 bg-white col-md-6 offset-md-3"
          >
            <div
              className="request-header d-flex flex-row justify-content-between rounded border px-2 py-2 bg-white"
            >
              <strong
                className="font-weight-bold"
              >
                fullname
              </strong>
              <Button
                buttonClass="view-btn rounded text-white bg-success"
                buttonName="view"
                onClick={[Function]}
              />
            </div>
            <div
              className="request-content d-none p-3"
            >
              <dl
                className="request-detail-dl"
              >
                <div
                  className="request-detail-div"
                >
                  <dt
                    className="status-term"
                  >
                    STATUS
                  </dt>
                  <dd
                    className="status-def"
                  >
                    resolved
                  </dd>
                </div>
              </dl>
            </div>
          </div>,
        ]
      }
      divClass="row py-5"
    />
  </div>
</div>
`;
